### PR TITLE
Allow pflag creation for slices

### PR DESCRIPTION
**Pull request by NickyMateev**
# Allow pflag creation for slices

## Motivation

There could be situations where slices are part of SM's configuration and corresponding pflags should be able to be created based on that.

## Approach

Remove the restriction for slices.

## Pull Request status

For example:

* [x] Initial implementation

### CODE REVIEW

**Comment by dpanayotov** [comment]
Can you add some tests in env_test that use slices

File: pkg/env/env.go [hunk 1]
```diff
@@ -77,7 +77,12 @@ func CreatePFlags(set *pflag.FlagSet, value interface{}) {
 
 	for i, parameter := range parameters {
 		if set.Lookup(parameter.Name) == nil {
-			set.Var(&flag{value: parameter.DefaultValue}, parameter.Name, descriptions[i])
+			switch val := parameter.DefaultValue.(type) {
+			case []string:
```
Comment (dpanayotov) [inline pkg/env/env.go]: remove this and add the following to the `flag.Type` method:
```golang
if kind == reflect.Slice {
	return fmt.Sprintf("%sSlice", typeOfValue.Elem().Name())
}
```

File: pkg/env/helpers.go [hunk 2]
```diff
@@ -126,5 +126,5 @@ func buildDescriptionTreeWithParameters(value interface{}, tree *descriptionTree
```
Comment (dpanayotov) [inline pkg/env/helpers.go]: maybe add 
```golang
if name == "-" || name == ",squash" {
	continue
}
```
on line 112

File: pkg/env/env.go (hunk 1 above)
Comment (NickyMateev) [reply pkg/env/env.go]: This opens the possibility of calling .Set() on the flag, which won't reflect the way a stringSlice flag would behave, could cause issues in the future. What do you think?

**Pull request merged**